htsget-config/src/tls/mod.rs

```diff
@@ -141,51 +144,39 @@ impl KeyPairScheme for Option<&TlsServerConfig> {
   }
 }
 
-/// Load a private key from a file. Supports RSA, PKCS8, and Sec1 encoded keys.
-pub fn load_key<P: AsRef<Path>>(key: P) -> Result<PrivateKey> {
-  let mut key_reader = BufReader::new(
-    File::open(key).map_err(|err| IoError(format!("failed to open key file: {}", err)))?,
-  );
+/// Loads the first private key from a file. Supports RSA, PKCS8, and Sec1 encoded EC keys.
+pub fn load_key<P: AsRef<Path>>(key_path: P) -> Result<PrivateKeyDer<'static>> {
+  let mut key_reader = BufReader::new(File::open(key_path)?);
 
   loop {
-    match read_one(&mut key_reader)
-      .map_err(|err| ParseError(format!("failed to parse private key: {}", err)))?
-    {
-      Some(rustls_pemfile::Item::RSAKey(key)) => return Ok(PrivateKey(key)),
-      Some(rustls_pemfile::Item::PKCS8Key(key)) => return Ok(PrivateKey(key)),
-      Some(rustls_pemfile::Item::ECKey(key)) => return Ok(PrivateKey(key)),
+    match read_one(&mut key_reader)? {
+      Some(Pkcs1Key(key)) => return Ok(PrivateKeyDer::from(key)),
+      Some(Pkcs8Key(key)) => return Ok(PrivateKeyDer::from(key)),
+      Some(Sec1Key(key)) => return Ok(PrivateKeyDer::from(key)),
+      // Silently disregard unknown private keys.
```

Comment:
🤔 ... not sure we want to be so silent here?

Reply:
I think it's okay. This is what `rustls_pemfile` does by default within the `certs` function. The .pem file can have multiple certs/keys in it, so it's possible that a user wants to have the same file contain the certificate and key. This means that it shouldn't be an error to read a certificate and key from the same file.